### PR TITLE
PluginPage: Update example to use new PluginPage and react router switch 

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Use react router instead of query params
* [x] Use new PluginPage component, including list and item page
* [x] Make it backward compatible with older version of Grafana that does not have PluginPage (or when topnav is disabled)

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports.getWebpackConfig = (config, options) => {
+  return {
+    ...config,
+    externals: [
+      ...config.externals,
+      'react-router-dom',
```
Comment (jackw) [inline webpack.config.js:6]: Shall we remove this file in favour of pinning `@grafana/toolkit` to 9.1.3 now that grafana/grafana/pull/54519 is merged and released?

File: src/PluginPage.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { PluginPageProps, PluginPage as RealPluginPage, config } from "@grafana/runtime";
```
Comment (jackw) [inline src/PluginPage.tsx]: I'm not sure how you managed to get this to build Torkel. The version of `runtime` in declared in `package.json` doesn't have these imports available. Should we bump the `@grafana` package versions to 9.1.x? Not sure how else we're going to get this to work. 🤔 

```
  ERROR in /Users/jackwestbrook/Projects/grafana-plugins/grafana-starter-app/src/pages/CanvasPage.tsx
  ERROR in /Users/jackwestbrook/Projects/grafana-plugins/grafana-starter-app/src/pages/CanvasPage.tsx(2,25):
  TS2305: Module '"../../node_modules/@grafana/runtime"' has no exported member 'PluginPageLayout'.

  ERROR in /Users/jackwestbrook/Projects/grafana-plugins/grafana-starter-app/src/PluginPage.tsx
  ERROR in /Users/jackwestbrook/Projects/grafana-plugins/grafana-starter-app/src/PluginPage.tsx(1,10):
  TS2305: Module '"../node_modules/@grafana/runtime"' has no exported member 'PluginPageProps'.

  ERROR in /Users/jackwestbrook/Projects/grafana-plugins/grafana-starter-app/src/PluginPage.tsx
  ERROR in /Users/jackwestbrook/Projects/grafana-plugins/grafana-starter-app/src/PluginPage.tsx(1,27):
  TS2305: Module '"../node_modules/@grafana/runtime"' has no exported member 'PluginPage'.
```

Comment (torkelo) [reply src/PluginPage.tsx]: @jackw you need to yarn link things :)

Comment (torkelo) [reply src/PluginPage.tsx]: @jackw is there a way to specify in package.json to reference canary (main) builds of npm packages?

Comment (jackw) [reply src/PluginPage.tsx]: Ah of course. So we could do `"@grafana/runtime": "canary"` but it's still gonna add a "specific" version to the lock file any how. Might be nicer to go with the last canary release `9.2.0-79773pre` for now so it's obvious what version we're using.

**Review by leventebalogh — COMMENTED** [review]
I think this logic is a good way forward! 👍 

On the other hand we were deprecating this repo in favour of https://github.com/grafana/grafana-plugin-examples, so I have moved the code over to there in https://github.com/grafana/grafana-plugin-examples/pull/98 (that option uses another experimental utility package that we were talking about to make applying backwards-compatibility logic like this easier), what do you think?

**Pull request closed without merging**